nd4j/nd4j-parameter-server-parent/nd4j-parameter-server-node/src/main/java/org/nd4j/parameterserver/distributed/v2/chunks/impl/FileChunksTracker.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2015-2018 Skymind, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.nd4j.parameterserver.distributed.v2.chunks.impl;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.val;
+import org.nd4j.linalg.exception.ND4JIllegalStateException;
+import org.nd4j.linalg.primitives.AtomicBoolean;
+import org.nd4j.linalg.util.SerializationUtils;
+import org.nd4j.parameterserver.distributed.v2.chunks.ChunksTracker;
+import org.nd4j.parameterserver.distributed.v2.chunks.VoidChunk;
+import org.nd4j.parameterserver.distributed.v2.messages.VoidMessage;
+
+import java.io.*;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * File-based implementation of ChunksTracker
+ */
+public class FileChunksTracker<T extends VoidMessage> implements ChunksTracker<T> {
+    @Getter
+    private final String originId;
+
+    private final int numChunks;
+
+    private Map<Integer, AtomicBoolean> map = new ConcurrentHashMap<>();
+
+    private File holder;
+
+    private final long size;
+
+    public FileChunksTracker(VoidChunk chunk) {
+        originId = chunk.getOriginalId();
+        numChunks = chunk.getNumberOfChunks();
+        size = chunk.getTotalSize();
+
+        try {
+            holder = File.createTempFile("FileChunksTracker", "Message");
```

Comment:
Do we want configurable location?
i.e., for VMs etc with slow system  HDD but fast SSD available

Reply:
Probably yes... Try configurable location first, fallback to default temp second.
But user can just redefine java temp dir as well...